toolchain/check/generic.cpp

```diff
@@ -177,6 +158,20 @@ class RebuildGenericConstantInEvalBlockCallbacks : public SubstInstCallbacks {
     return result.value();
   }
 
+  auto ReuseUnchanged(SemIR::InstId orig_inst_id) const
+      -> SemIR::InstId override {
+    auto inst = context_.insts().Get(orig_inst_id);
+    CARBON_CHECK(
+        inst.Is<SemIR::BindSymbolicName>() ||
+            inst.Is<SemIR::SymbolicBindingPattern>(),
+        "Instruction {0} has symbolic constant value but no symbolic operands",
+        inst);
+
+    // Rebuild the instruction anyway so that it's included in the eval block.
+    // TODO: Can we just reuse the instruction in this case?
```

Comment:
How are we going to answer this question?

Reply:
I'm planning to look into this as part of the work to retain the (non-canonical) computation of the constant in the eval block.